packages/stylus/index.bzl

```diff
@@ -0,0 +1,43 @@
+"""
+Support running the stylus processor as a Bazel rule.
+"""
+
+def _stylus_binary(ctx):
+    src = ctx.file.src
+
+    # We want foo.styl to produce foo.css
+    output_name = src.basename[:-5] + ".css"
+    css_output = ctx.actions.declare_file(output_name)
+    map_output = ctx.actions.declare_file(output_name + ".map")
+    ctx.actions.run(
+        outputs = [css_output, map_output],
+        inputs = [src] + ctx.files.deps,
+        executable = ctx.executable._compiler,
+        arguments = [
```

Comment:
Probably we should enable `compress` and `resolve-url`.

Also, we should probably a way to pass the `include`

More info: http://stylus-lang.com/docs/executable.html

Reply:
adding a test for `resolve-url` shows an interesting consequence, TAL. Same problem as sourcemaps...

How would users want to use `include`? seems like that should just come from the path of the `deps`?

Reply:
I was thinking more something similar to the `sass_binary` and how they handle Sass include paths.

They have an additional attribute for this, I also think that with this approach it might be might be more performant  since imports resolutions will be limited to the fine grained include paths installed of the entire `deps`.

https://github.com/bazelbuild/rules_sass#implicit-output-targets

In future also, we might want to make `compress` and `sourcemap` configurable by the user.

Reply:
Asked Jeremy, he says this attribute only exists on sass_binary for legacy reasons and it would be better not to include it for new code.

Added user-config for compress and sourcemap.

Reply:
What is the new alternative/suggested way without include paths?

As if users want to import let's say `bootstrap` in their scss file. They will need to do a full import without include paths?

```css
@import '../../node_modules/bootstrap/scss/functions'
```

Some context https://github.com/angular/angular/issues/31619#issuecomment-512707874

Reply:
I would expect that to be written `@import 'bootstrap/scss/functions'` - meaning the node_modules should always be in the sass include path. I didn't check if that actually works though.

Reply:
Afaik preprocess include paths are empty by default.

Reply:
yeah so I think we should add a test for this, and hard-code correct paths. I'm not sure users should have control over something that affects paths

Reply:
Yeah some tests here are mandatory. I mean we can start with hard coded paths and evaluate the option to add paths publicly if needs be.